### ISSUE TITLE
Unattended Install non-default irods home results in error

### ISSUE THREAD

**Issue by alexander-haller** [irods/irods#6480]
### iRODS Version, OS and Version

apt show irods-server :
Version: 4.3.0-1~bullseye
Maintainer: iRODS Consortium <packages@irods.org>

OS: Debian 11.4

### What did you try to do?

Unattended Install (with ansible):
shell: "python3 /var/lib/irods/scripts/setup_irods.py --json_configuration_file /root/irods_provider_unattended_installation.json.input

irods_provider_unattended_installation.json.input is mostly taken from the docs here https://docs.irods.org/4.3.0/getting_started/installation/#input-file-for-catalog-service-provider

i changed irods_cwd, irods_home, to /testzone/srv/rods (from)  /testzone/home/rods

### Expected behavior

install runs trough without problems

### Observed behavior (including steps to reproduce, if applicable)

`Error encountered running setup_irods:\nTraceback (most recent call last):\n  File \"/var/lib/irods/scripts/setup_irods.py\", line 515, in main\n    setup_server(irods_config,\n  File \"/var/lib/irods/scripts/setup_irods.py\", line 150, in setup_server\n    test_put(irods_config)\n  File \"/var/lib/irods/scripts/setup_irods.py\", line 180, in test_put\n    raise IrodsError('Post-install test failed. Please check your configuration.')\nirods.exceptions.IrodsError: Post-install test failed. Please check your configuration.\u001b[0m", "stderr_lines": ["\u001b[1;31mError encountered running setup_irods:", "Traceback (most recent call last):", "  File \"/var/lib/irods/scripts/setup_irods.py\", line 515, in main", "    setup_server(irods_config,", "  File \"/var/lib/irods/scripts/setup_irods.py\", line 150, in setup_server", "    test_put(irods_config)", "  File \"/var/lib/irods/scripts/setup_irods.py\", line 180, in test_put", "    raise IrodsError('Post-install test failed. Please check your configuration.')", "irods.exceptions.IrodsError: Post-install test failed. Please check your configuration.`

```
irods@dev-storage-001:  ~$ ils
remote addresses: 127.0.1.1 ERROR: lsUtil: srcPath /testzone/srv/rods does not exist or user lacks access permission
irods@dev-storage-001:  ~$ /usr/sbin/irodsTestPutGet
Error: cannot open data object for writing ["/testzone/srv/rods/post_install_test.1658283396113650.txt"]

irods@dev-storage-001:  ~$ l /var/lib/irods/
..
no Vault dir
..
```

If I change the var to  /testzone/home/rods the setup runs fine.

I notice that /var/lib/irods/Vault exists and has has drwxr-x--- perms and the correct subfolders /var/lib/irods/Vault/home/rods

```
tree Vault/
Vault/
└── home
    └── rods
```

**Comment by korydraughn** [comment]
That is expected.

iRODS does not support changing the home collection during setup at this time. The setup script assumes certain things will be configured a particular way. This allows us to make assertions regarding correctness. After setup completes, you're free to change the home collection. To do that, you'd do the following:
1. Create a new collection using `imkdir`.
2. Open `/var/lib/irods/.irods/irods_environment.json` and set `irods_home` to the absolute (logical) path of the new collection.

With that said, allowing this in the future is a possibly.

Is this something you need?
Can you explain the motivation behind this?

**Comment by alexander-haller** [comment]
Glad to hear that is is not a bug.

If this is on purpose it would be helpful to mark in the documentation the fields that can be changed in the first automated install.
Since you provide an input file here https://docs.irods.org/4.3.0/getting_started/installation/#input-file-for-catalog-service-provider it is not clear (to me) what can be changed and what can not.
And the error from setup_irods.py  is also not indicative to the root cause.

Changing this later is ok for me - thanks.

> Q. Is this something you need?

It's more I expect this to work given the example in the docs. As long as I know what works I can change it later. But also I think it would be nicer if it works automated in a unified "one step" way.

> Q. Can you explain the motivation behind this?

Mostly I wanted to mirror my os path setup on the server since I do not use the /home dirs on this system. The system provides only a service and has no "human" users". So my service account has /srv/<account> as the home dir.

**Comment by korydraughn** [comment]
Yes, I agree with your comments. We'll work on improving the documentation around this.
